### PR TITLE
[ROU110] - Add a new rule for disallowing .save() without update fields or comments justifying it

### PR DESCRIPTION
**Pull request by tomharel**
This PR adds a new lint rule `ROU110` for disallowing `.save()` without update fields or comments justifying it.

### CODE REVIEW

File: flake8_routable.py [hunk 1]
```diff
@@ -349,14 +351,51 @@ def rename_migrations(self) -> None:
...
+    def disallow_no_update_fields_save(self) -> None:
+        """.save() must be called with update_fields."""
+        reported = set()
+        single_line_save = re.compile(r".+(\.save\(.*)")
+        allowed_comments = [
+            "# TODO: needs fix",
+            "# file save",
+            "# form save",
+            "# ledger save",
+            "# multiline with update_fields",
```
Comment (hurgleburgler) [inline flake8_routable.py:363]:
```suggestion
            "# multi-line with update_fields",
```

File: tests/test_flake8_routable.py [hunk 2]
```diff
@@ -536,6 +536,128 @@ def test_incorrect_import_from_tests(self):
...
 
+class TestROU110:
+    SAVE_WITH_UPDATE_FIELDS = """from app.models import Model
+instance = Model(id="123", name="test")
+instance.save(update_fields=["id", "name"])
+"""
+
+    SAVE_MULTILINE_WITH_UPDATE_FIELDS_FLAG = """from app.models import Model
+instance = Model(id="123", name="test")
+instance.save(  # multiline with update_fields
```
Comment (hurgleburgler) [inline tests/test_flake8_routable.py:547]:
```suggestion
instance.save(  # multi-line with update_fields
```

Comment (justmobilize) [reply tests/test_flake8_routable.py]: fixed here: https://github.com/routablehq/flake8-routable/pull/18/commits/4e832c01419e6fc72bcaab52a8dc5274b5a2217b

File: flake8_routable.py (hunk 1 above)
Comment (justmobilize) [reply flake8_routable.py]: fixed here: https://github.com/routablehq/flake8-routable/pull/18/commits/4e832c01419e6fc72bcaab52a8dc5274b5a2217b

**Review by henriquebastos — APPROVED** [review]
Nice. LGTM.

**Review by hurgleburgler — APPROVED** [review]

**Pull request merged**